crates/viewer/re_space_view_spatial/src/visualizers/points2d.rs

```diff
@@ -211,53 +211,62 @@ impl VisualizerSystem for Points2DVisualizer {
         line_builder
             .radius_boost_in_ui_points_for_outlines(SIZE_BOOST_IN_POINTS_FOR_POINT_OUTLINES);
 
-        super::entity_iterator::process_archetype::<Self, Points2D, _>(
+        super::entity_iterator::process_archetype2::<Self, Points2D, _>(
             ctx,
             view_query,
             context_systems,
             |ctx, spatial_ctx, results| {
-                use re_space_view::RangeResultsExt as _;
+                use re_space_view::RangeResultsExt2 as _;
 
-                let resolver = ctx.recording().resolver();
-
-                let positions = match results.get_required_component_dense::<Position2D>(resolver) {
-                    Some(positions) => positions?,
-                    _ => return Ok(()),
+                let Some(all_position_chunks) = results.get_required_chunks(&Position2D::name())
+                else {
+                    return Ok(());
                 };
 
-                let num_positions = positions
-                    .range_indexed()
-                    .map(|(_, positions)| positions.len())
-                    .sum::<usize>();
+                let num_positions = all_position_chunks
+                    .iter()
+                    .flat_map(|chunk| chunk.iter_primitive_array::<2, f32>(&Position2D::name()))
```

Comment:
it's a bit unfortunate how easy it is to get this wrong here: We first specify `Position2D`'s name, then have to know this is two floats and then cast it to `Position2D`. If any of these steps are wrong, we get into a weird state.

Reply:
Same remark as above: you don't really get into a weird state most of the time, you just insta panic. And again you have to be actively modifying this code etc.